docs/howtos/lr_schedule.rst

```diff
@@ -0,0 +1,74 @@
+Managing Parameters and State
+=============================
+
+We will show you how to...
+
+* define a learning rate schedule
+* train a simple model using that schedule
+
+.. code-block:: python
+  
+  def create_triangular_schedule(lr_min, lr_max, steps_per_cycle):
+  """Return fn from epoch to LR, linearly interpolating between `lr_min`->`lr_max`->`lr_min`."""
+    top = (steps_per_cycle + 1) // 2
+    def learning_rate_fn(step):
+      cycle_step = step % steps_per_cycle
+      if cycle_step < top:
+        lr = lr_min + cycle_step/top * (lr_max - lr_min)
+      else:
+        lr = lr_max - ((cycle_step - top)/top) * (lr_max - lr_min)
+      return lr
+    return learning_rate_fn
+
+This example shows how to create a simple cyclical learning rate scheduler, the triangular scheduler.
+
+To use the schedule one must simply create a learning rate function by passing the hyperparameters to the 
+create_triangular_schedule function and then use that function to compute the learning rate for your updates.
+For example using this schedule on MNIST would require changing the train_step function
+
+.. code-block:: python
+  
+  def train_step(optimizer, batch, learning_rate_fn):  
+    def loss_fn(params):
+      logits = CNN().apply({'params': params}, batch['image'])
+      loss = cross_entropy_loss(logits, batch['label'])
+      return loss, logits
+    
+    grad_fn = jax.value_and_grad(loss_fn, has_aux=True)
+    (_, logits), grad = grad_fn(optimizer.target)
+    step = optimizer.state.step
+    lr = learning_rate_fn(step)
+    optimizer = optimizer.apply_gradient(grad, {"learning_rate":lr})
+    metrics = compute_metrics(logits, batch['label'])
+    return optimizer, metrics
+
+And the train_epoch function:
```

Comment:
Could you rewrite this codeblock using the codediff directive as well, as I did above?

Reply:
Is there a way to get syntax highlighting with codediff?

Reply:
Yes, you simply append #! to each line you want to highlight.